### ISSUE TITLE
Can not delete empty experiments

### ISSUE THREAD

**Issue by ZZZanoY** [genomizer/genomizer-server#331]
When trying to delete an experiment that does not contain any files, the response is 500 temporary problem with database.

**Comment by rednatto** [comment]
````NEW EXCHANGE: DELETE /experiment/TETS
Trying to authenticate token 3dbe9a7f-1975-453f-b945-03cc47546f1e...
User testadmin authenticated successfully.
Request body: 
{}
Deletion of experiment TETS didn't work, reason: ERROR: update or delete on table "experiment" violates foreign key constraint "fkey_expid" on table "file"
  Detail: Key (expid)=(TETS) is still referenced from table "file".
Response body: 
{"message":"Deletion of experiment TETS didn't work due to temporary problems with the database."}

END OF EXCHANGE````

**Comment by tastyburger** [comment]
When an experiment has files which are marked as `Done`, the experiment cannot be deleted directly. I presume, those files have to be deleted first somehow to proceed with the deletion of the experiment. The problem is that some files never complete uploading and the status remains as `In Progress` and this creates a reference to the experiment and hence the experiment cannot be deleted to not violate the referential integrity.

**Comment by unicornmaster** [comment]
I'll look in to it

**Comment by unicornmaster** [comment]
The problem was that files marked 'In progress' where not deleted on removal of an experiment.
It should be fixed now

**Comment by unicornmaster** [comment]
Fixed in #354
